cep-purls.md

```diff
@@ -0,0 +1,192 @@
+  }
+}
+```
+
+If the `purls` field is not present in the `repodata.json` record, it means no `purls` information is stored with the record. Then, a fallback mechanism (i.e., read `purls.json`) should be used to acquire the PURL information.
+
+In sharded repodata ([CEP 16](cep-0016.md)), PURLs can be specified similar to [CEP 21](cep-0021.md).
+
+### Patching
+
+With the purls part of the repodata we propose to also allow repodata patching this field. The original purls can still be extracted from the packages in `index.json`. We do not see a reason how the PURL information is different from other patchable information stored in the repodata (like the dependencies).
+
+To facilitate these patches, the `patch_instructions_version` in `patch_instructions.json` files is incremented to `3`. A `patch_instructions_version: 3` file MAY contain a `purls: ...` field in the patch instructions that MUST be used to patch generated the `purls` entry in the (sharded) repodata (or the `purls.json`).
+
+```json
+{
+  "patch_instructions_version": 3,
+  "packages": {
+    "package-version-build.tar.bz2": {
+      ...,
+      "purls": ["pkg:pypi/pinject@0.14.1"]
+    },
+    ...
+  }
+}
+```
+
+### PURL of a conda packages
+
+Conda packages itself should have a PURL as well.
+This makes it possible for CVE Numbering Autorities (CNA) to publish vulnerabilities for conda packages.
+A package `pinject-0.14.1-pyh29332c3_1.conda` published on `conda-forge` should have the PURL `pkg:conda/conda-forge/pinject@0.14.1?build=pyh29332c3_1&platform=noarch`.
+Packages can also specify a custom `repository_url`, for example `pkg:conda/custom-channel/my-package@0.1.0?build=pyh29332c3_1&platform=noarch&repository_url=prefix.dev`.
+
+Tools that generate packages like rattler-build and conda-build should also be able to inject a PURL at build time via CLI flags.
+For this, v1-recipes can use the jinja syntax from the recipes.
+
+```bash
+rattler-build build -r recipe/ \
+    --append-purl-pattern \
+    'pkg:conda/conda-forge/${{ PACKAGE_NAME }}@${{ PACKAGE_VERSION }}?build=${{ BUILD_NUMBER }}'
+```
+
+Variables that are available in the build process (like `PACKAGE_NAME`, `PACKAGE_VERSION` and `BUILD_NUMBER`) can be specified here.
+
+## Motivation
+
+Conda packages can mostly repackage packages from other ecosystems.
+Conda-forge and other channels famously repackages a lot of PyPI packages.
+However, without actually downloading the conda package and inspecting its contents there is no reliable way to know whether a certain conda package is a repackaged package and which package it repackages.
+
+Tools like pixi or conda-lock try to combine conda and PyPI packages through heuristics. This doesn't work deterministically as package names between the two indices may differ.
+
+Its hard to use open-source vulnerability databases because they often do not contain conda packages.
+Using the PURL standard allows us to link vulnerabilities from other ecosystems to conda package.
```

Comment:
I would move this above Specification so it's clear _why_ we are doing this. It would also need some more details about the intended usage and benefits, since there are many!